Deadline Improvments

davidhennemann opened the pull request:
This Branch includes:
- deletation of unnecessary code
- update button, will update flags of existing mails
- loadOlderMails will work flawlessly now + tests
- Changes on server will be reflected back to client :)
- Warning if mail was moved on the remote
- correct indent for mail folder hierarchy in context menu
- deletation works fine in `ICLoginDialog` now

davidhennemann committed:
Removing unnecessary code.

davidhennemann committed:
After pressing update flags will be updated. This also includes mails, which are already present in storage.

davidhennemann committed:
Add test for retriveFlagsOnly.

davidhennemann committed:
Rename method arguments of fetchHeaders.

davidhennemann committed:
Refactor fetchHeadersFrom:To: into useful method + add tests. Simple Refactoring.

ClayGee commented on packages/IMAPClient-Core.package/ICEndPoint.class/instance/fetchMailsBeginningAt.number..st:
```diff
@@ -0,0 +1,24 @@
+	| start end possibleUID numberToFetch|
+	
+	"aBeginUID is the last known UID of the folder. numberOfMessages is the number
+	 of messages which should be fetched starting at this UID"
```
aBeginUID instead of UID for better understanding <3

davidhennemann committed:
Check whether the server moved mails and reflect changes back to client.

ClayGee committed:
Fix Bug: Dont load older mails beginning at first uid in sorted collection, because collection is sorted by date and not by uid.

davidhennemann committed:
Addresses several minor bugs:
	- deselect mail after folder change
	- correct indentation for mail folders
	. context-menu only openable for valid mails

davidhennemann committed:
Deletation of last character in LoginDialog works now.

davidhennemann committed:
Display a warning if mail was changed on server.
